### PR TITLE
Server tests

### PR DESCRIPTION
**Pull request by sabath91**

### CODE REVIEW

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package com.epam.ships.server.gamestates;
+
+import com.epam.ships.server.CommunicationBus;
+import org.testng.annotations.Test;
+
+import static org.mockito.Mockito.*;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+@Test
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java]: Add a blank line:)

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+public class GameEndWithWalkoverStateTest {
+
+  public void shouldProcessWinState() {
+    //given
+    CommunicationBus communicationBus = mock(CommunicationBus.class);
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java]: Create a private field for CommunicationBus and mock it with BeforeMethod if you need it refreshed before each test or with setUp with you don't

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    //when
+    GameEndWithWalkoverState gameEndWithWalkover= new GameEndWithWalkoverState(communicationBus);
+    GameState gameState = gameEndWithWalkover.process();
+
+    //then
+    assertEquals(GameEndWithWalkoverState.class, gameState.getClass());
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java]: Class based assertions aren't elegant, possible to improve it?

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    verify(communicationBus, times(1)).stop();
+  }
+
+  public void shouldReturnFalseWhenShouldBeContinuedIsCall() {
+    //given
+    CommunicationBus communicationBus = mock(CommunicationBus.class);
+    //when
+    GameEndWithWalkoverState gameEndWithWalkover= new GameEndWithWalkoverState(communicationBus);
+    boolean result = gameEndWithWalkover.shouldBeContinued();
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java]: Move this line to //then or even put gameEndWithWalkover.shouldBeContinued() into assertFalse

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+
+@Test
+public class GameEndWithWinStateTest {
+
+  public void shouldProcessWinState() {
+    //given
+    CommunicationBus communicationBus = mock(CommunicationBus.class);
+    TurnManager turnManager = mock(TurnManager.class);
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java]: Create private fields for CommunicationBus, TurnManager and mock them with BeforeMethod if you need them refreshed before each test or with setUp with you don't

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    //when
+    GameEndWithWinState gameEndWithWinState = new GameEndWithWinState(communicationBus, turnManager);
+    GameState gameState = gameEndWithWinState.process();
+
+    //then
+    assertEquals(GameEndWithWinState.class, gameState.getClass());
+    verify(communicationBus, times(1)).stop();
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java]: Move one assertion to a dedicated test

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import static org.testng.Assert.assertFalse;
+@Test
+public class GameEndWithWalkoverStateTest {
+
+  public void shouldProcessWinState() {
+    //given
+    CommunicationBus communicationBus = mock(CommunicationBus.class);
+
+    //when
+    GameEndWithWalkoverState gameEndWithWalkover= new GameEndWithWalkoverState(communicationBus);
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java]: Move this to given

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    //then
+    assertEquals(GameEndWithWalkoverState.class, gameState.getClass());
+    verify(communicationBus, times(1)).stop();
+  }
+
+  public void shouldReturnFalseWhenShouldBeContinuedIsCall() {
+    //given
+    CommunicationBus communicationBus = mock(CommunicationBus.class);
+    //when
+    GameEndWithWalkoverState gameEndWithWalkover= new GameEndWithWalkoverState(communicationBus);
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java]: Move this to given

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+@Test
+public class GameEndWithWinStateTest {
+
+  public void shouldProcessWinState() {
+    //given
+    CommunicationBus communicationBus = mock(CommunicationBus.class);
+    TurnManager turnManager = mock(TurnManager.class);
+
+    //when
+    GameEndWithWinState gameEndWithWinState = new GameEndWithWinState(communicationBus, turnManager);
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java]: Move this to given

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+    //then
+    assertEquals(GameEndWithWinState.class, gameState.getClass());
+    verify(communicationBus, times(1)).stop();
+  }
+
+  public void shouldReturnFalseWhenShouldBeContinuedIsCall() {
+    //given
+    CommunicationBus communicationBus = mock(CommunicationBus.class);
+    TurnManager turnManager = mock(TurnManager.class);
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java]: Create private fields for CommunicationBus, TurnManager and mock them with BeforeMethod if you need them refreshed before each test or with setUp with you don't

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    //when
+    GameEndWithWinState gameEndWithWinState = new GameEndWithWinState(communicationBus, turnManager);
+    boolean result = gameEndWithWinState.shouldBeContinued();
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java]: Move this line to //then or even put gameEndWithWalkover.shouldBeContinued() into assertFalse

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+    assertEquals(GameEndWithWinState.class, gameState.getClass());
+    verify(communicationBus, times(1)).stop();
+  }
+
+  public void shouldReturnFalseWhenShouldBeContinuedIsCall() {
+    //given
+    CommunicationBus communicationBus = mock(CommunicationBus.class);
+    TurnManager turnManager = mock(TurnManager.class);
+    //when
+    GameEndWithWinState gameEndWithWinState = new GameEndWithWinState(communicationBus, turnManager);
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java]: Move this to given

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+    boolean result = gameEndWithWinState.shouldBeContinued();
+    //then
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java]: rename to //when - then

File: server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java [hunk 14]
```diff
@@ -28,4 +31,14 @@ public void shouldProcess(){
     assertNotNull(fleetPlacementState);
   }
 
+  public void shouldBeContinuedShouldReturnTrueByDefault(){
+    //given
+    CommunicationBus communicationBus = mock(CommunicationBus.class);
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java]: Create a private field for CommunicationBus and mock it with BeforeMethod if you need it refreshed before each test or with setUp with you don't

File: server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java [hunk 15]
```diff
@@ -28,4 +31,14 @@ public void shouldProcess(){
     assertNotNull(fleetPlacementState);
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java]: Move to  a dedicated test

File: server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java [hunk 16]
```diff
@@ -28,4 +31,14 @@ public void shouldProcess(){
...
   }
 
+  public void shouldBeContinuedShouldReturnTrueByDefault(){
+    //given
+    CommunicationBus communicationBus = mock(CommunicationBus.class);
+    //when
+    GameState gameState = new WaitingForPlayersState(communicationBus);
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java]: Move to given

File: server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java [hunk 17]
```diff
@@ -28,4 +31,14 @@ public void shouldProcess(){
...
+    boolean result = gameState.shouldBeContinued();
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java]: Move this line to //then or even put gameState.shouldBeContinued() into assertTrue, and then rename //then to //when - then

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java (hunk 1 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+  private WrappedClient wrappedClient;
+
+  @BeforeMethod
+  void before(){
+    initMocks(this);
+  }
+
+
+  public void shouldKeepStateWhenNoWinner() {
+    Fleet fleet2 = Fleet.ofShips(Arrays.asList(Ship.ofMasts(Mast.ofIndex("4"), Mast.ofIndex("5"))));
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: Rename fleet2 to secondFleet

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java (hunk 2 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java [hunk 19]
```diff
@@ -0,0 +1,100 @@
...
+    List<Fleet> fleetList = Arrays.asList(fleet, fleet2);
+    processShot();
+    //when
+    PlayState playState = new PlayState(communicationBus, fleetList);
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: Move to given

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java (hunk 3 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java]: No. I have to check it in this way.

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java [hunk 20]
```diff
@@ -0,0 +1,100 @@
...
+    GameState gameState = playState.process();
+    //then
+    assertEquals(PlayState.class, gameState.getClass());
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: Class based assertions aren't elegant, possible to improve?

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java [hunk 21]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  public void shouldProcessWin() {
+    //given
+    Fleet fleet2 = Fleet.ofShips(Arrays.asList(Ship.ofMasts(Mast.ofIndex("4"))));
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: Rename fleet2 to secondFleet, anotherFleet or anything meaningful without a number in its name :P

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java [hunk 22]
```diff
@@ -0,0 +1,100 @@
...
+    processShot();
+    //when
+    PlayState playState = new PlayState(communicationBus, fleetList);
+    GameState gameState = playState.process();
+    //then
+    assertEquals(GameEndWithWinState.class, gameState.getClass());
+  }
+
+  public void shouldProcessWalkover() {
+    Fleet fleet2 = mock(Fleet.class);
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: Rename fleet2 to secondFleet, anotherFleet or anything meaningful without a number in its name :P

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java [hunk 23]
```diff
@@ -0,0 +1,100 @@
...
+    List<Fleet> fleetList = Arrays.asList(fleet, fleet2);
+    processShot();
+    //when
+    PlayState playState = new PlayState(communicationBus, fleetList);
+    GameState gameState = playState.process();
+    //then
+    assertEquals(PlayState.class, gameState.getClass());
+  }
+
+  public void shouldProcessWin() {
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: I'd replace "process" with SwitchTo or smth else more related to a result

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java [hunk 24]
```diff
@@ -0,0 +1,100 @@
...
+    //given
+    Fleet fleet2 = Fleet.ofShips(Arrays.asList(Ship.ofMasts(Mast.ofIndex("4"))));
+    List<Fleet> fleetList = Arrays.asList(fleet, fleet2);
+    processShot();
+    //when
+    PlayState playState = new PlayState(communicationBus, fleetList);
+    GameState gameState = playState.process();
+    //then
+    assertEquals(GameEndWithWinState.class, gameState.getClass());
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: Class based assertions aren't elegant, possible to improve?

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java [hunk 25]
```diff
@@ -0,0 +1,100 @@
...
+    assertEquals(PlayState.class, gameState.getClass());
+  }
+
+  public void shouldProcessWin() {
+    //given
+    Fleet fleet2 = Fleet.ofShips(Arrays.asList(Ship.ofMasts(Mast.ofIndex("4"))));
+    List<Fleet> fleetList = Arrays.asList(fleet, fleet2);
+    processShot();
+    //when
+    PlayState playState = new PlayState(communicationBus, fleetList);
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: Move to given

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java [hunk 26]
```diff
@@ -0,0 +1,100 @@
...
+    Message shot = new MessageBuilder()
+        .withAuthor(Author.AUTO)
+        .withHeader(Header.CONNECTION)
+        .withStatement("END")
+        .build();
+    when(communicationBus.receive(sender)).thenReturn(shot);
+    when(communicationBus.getFirstClient()).thenReturn(sender);
+    when(communicationBus.getSecondClient()).thenReturn(wrappedClient);
+    //when
+    PlayState playState = new PlayState(communicationBus, fleetList);
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: Move to given

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java [hunk 27]
```diff
@@ -0,0 +1,100 @@
...
+    GameState gameState = playState.process();
+    //then
+    assertEquals(GameEndWithWalkoverState.class, gameState.getClass());
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: Class based assertions aren't elegant, possible to improve?

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java [hunk 28]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  private void processShot() {
+    when(communicationBus.receive(sender)).thenReturn(shot());
+    when(communicationBus.getFirstClient()).thenReturn(sender);
+    when(communicationBus.getSecondClient()).thenReturn(wrappedClient);
+  }
+
+  private Message shot() {
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: Rename to produceMessageWithShot (do not use just nouns in method names)

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java (hunk 4 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java]: I'll leave it separated

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java (hunk 5 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java (hunk 6 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java (hunk 7 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java (hunk 8 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWalkoverStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java (hunk 9 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java (hunk 10 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java (hunk 11 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java]: I'll leave it separated.

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java (hunk 12 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java (hunk 13 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/GameEndWithWinStateTest.java]: I'll leave when and then sections sepparated

File: server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java (hunk 14 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java]: Done

File: server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java (hunk 15 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java (hunk 16 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java (hunk 17 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/WaitingForPlayersStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java (hunk 20 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: I don't have an idea how even if it is possible.

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java (hunk 21 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java (hunk 22 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java (hunk 23 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java (hunk 24 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: No

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java (hunk 25 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: done.

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java (hunk 26 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java (hunk 27 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: as above

File: server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java (hunk 28 above)
Comment (sabath91) [reply server/src/test/java/com/epam/ships/server/gamestates/play/PlayStateTest.java]: done

File: server/src/test/java/com/epam/ships/server/gamestates/play/ShotHandlerTest.java [hunk 29]
```diff
@@ -87,7 +85,6 @@ public void shouldHandleFleetDefeat() {
     //given
     ShotHandler shotHandler = new ShotHandler(communicationBus, turnManager, fleets);
 
-
     //when
     Message message = shotAtIndex("11");
```
Comment (korotkevics) [inline server/src/test/java/com/epam/ships/server/gamestates/play/ShotHandlerTest.java]: firstMessage

Comment (korotkevics) [reply server/src/test/java/com/epam/ships/server/gamestates/play/ShotHandlerTest.java]: //when below

**Pull request merged**